sycl/source/detail/config.hpp

```diff
@@ -176,6 +176,85 @@ template <> class SYCLConfig<SYCL_PI_TRACE> {
   }
 };
 
+template <> class SYCLConfig<SYCL_PARALLEL_FOR_RANGE_ROUNDING_TRACE> {
+  using BaseT = SYCLConfigBase<SYCL_PARALLEL_FOR_RANGE_ROUNDING_TRACE>;
+
+public:
+  static bool get() {
+    static bool Initialized = false;
+    static bool Trace = false; // No tracing by default
+
+    // Check env var setting for tracing only once
+    if (Initialized)
+      return Trace;
+
+    const char *ValStr = BaseT::getRawValue();
+    Trace = ValStr != nullptr;
+    Initialized = true;
+    return Trace;
+  }
```

Comment:
we can make function shorter.
```suggestion
  static bool get() {
    static const char *ValStr = BaseT::getRawValue();
    return ValStr != nullptr;
  }
```

Reply:
The point of this overall change is to minimize calls to getenv. The function getRawValue always calls getenv (when ConfigFromEnvEnabled is true, which is the normal case). Therefore, the suggested change does not achieve the intended effect of minimizing calls to getenv.
The code as written, although appearing long-winded, is written that way for a reason. The flag Initialized is used to guard a call to getRawValue (i.e. getenv). Its use ensures that getenv is called only once per run of the program.
See existing code for SYCLConfig<SYCL_PI_TRACE>. The technique in the new code is borrowed from there.

Reply:
The local static variable is initialized only once. So it does not matter if getRawValue() call getenv() every time or not. The getRawValue() will be called only once (the first time the get() method is called).

Reply:
Yes, OK. Change made.